components/camel-vertx-kafka/camel-vertx-kafka-component/src/test/java/org/apache/camel/component/vertx/kafka/operations/VertxKafkaProducerOperationsTest.java

```diff
@@ -586,7 +586,7 @@ public void configure(Map<String, ?> configs, boolean isKey) {
 
         @Override
         public byte[] serialize(String topic, Object data) {
-            return stringSerializer.serialize(topic, data.toString());
+            return stringSerializer.serialize(topic, data != null ? data.toString() : null);
```

Comment:
Tests were failing before my changes with a `NullReferenceException`. Seems like `data` was `null` in some tests, causing the issue with the serializer